handlers.go

```diff
@@ -317,7 +317,11 @@ func (t *TricksterHandler) getVectorFromPrometheus(url string, params url.Values
 	// Unmarshal the prometheus data into another PrometheusMatrixEnvelope
 	err = json.Unmarshal(body, &pe)
 	if err != nil {
-		return pe, nil, fmt.Errorf("Prometheus vector unmarshaling error for URL %q: %v", url, err)
+		// If we get a scalar response, we just want to return the resp without an error
+		// this will allow the upper layers to just use the raw response
```

Comment:
Can you point me at the upper levels that still use the response if the JSON parsing failed?

For callers of `getVectorFromPrometheus()` I only found these:

- https://github.com/Comcast/trickster/blob/d7191ce7bc4cf222b1ddc5423d10981b3deb09d3/handlers.go#L615
- https://github.com/Comcast/trickster/blob/d7191ce7bc4cf222b1ddc5423d10981b3deb09d3/handlers.go#L782

In both cases, as far as I can see, only the HTTP status code of the response is ever used, not any scalar contents it might contain.

Reply:
@juliusv from my testing I see that everything besides that fast-forward request works. Its still not ideal as its missing a point, but this PR does put it back to the previous behavior (which was definitely less broken). I'm planning on focusing on #82 as the longer-term fix for this, since the old behavior was "good enough" (I didn't even notice it wasn't 100% right until working on this issue ;) )

Reply:
I guess it depends on whether you consider silently ignoring partial errors as "less broken" :) But fine, as long as it gets fixed properly later.

Reply:
Yup, we're keeping #82 open until the permanent fix is in